chapter2/tests/t3/chapter2-t3-velero-eks-diff.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+# Task 3 - find namespaces without the backup
+set -eo pipefail
+
+
+readonly VELERO_URL='https://gist.github.com/dmitry-mightydevops/016139747b6cefdc94160607f95ede74/raw/velero.yaml'
+readonly ALL_NAMESPACES_URL='https://gist.githubusercontent.com/dmitry-mightydevops/297c4e235b61982f21a0bbbf7319ac24/raw/kubernetes-namespaces.txt'
+
+
+# Function for printing error messages and exiting with a specific exit code
+# $1 - The exit code to use when exiting the script
+# $2 - The error message to print
+err() {
+  local code="$1"
+  shift
+  echo "[[ERROR]: $(date +'%Y-%m-%dT%H:%M:%S%z')]: ${*}" >&2
+  exit "${code}"
+}
+
+
+# Function for checking file existence and downloading it.
+# $1 - URL to the file
+# $2 - Var name to save
+get_file() {
+  local url="$1"
+  local save_to_var="$2"
+  if ! curl -sSL --head --request GET "${url}" | grep "HTTP/2\ 200" > /dev/null; then
+    err 255 "Error getting ${url}"
+  else
+    declare -g "$save_to_var=$(curl -sSL "${url}")"
+  fi
+}
+
+
+# Check if yq is installed.
+if ! command -v yq > /dev/null; then
+  err 255 'yq is not installed'
+fi
+
+
+get_file "${VELERO_URL}" "VELERO"
+get_file "${ALL_NAMESPACES_URL}" "all_namespaces"
```

Comment:
you should always avoid such functions. Your function should return string value that you consume in outer shell and convert into desired struct (in your case array). Line #30 about making the var is bad, makes code much harder to read and maintain.

so 
```
backup_namespaces=something ARG1 ARG2...ARGx
all_namespaces=something ARG1 ARG2...ARGx
```

then foreach to process and find the diff.